pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -440,12 +448,16 @@ def test_agg_relabel_non_identifier(self):
         )
         tm.assert_frame_equal(result, expected)
 
-    def test_duplicate_raises(self):
-        # TODO: we currently raise on multiple lambdas. We could *maybe*
-        # update com.get_callable_name to append `_i` to each lambda.
+    def test_duplicate_no_raises(self):
+        # GH 28426, if use same input function on same column,
+        # no error should raise
         df = pd.DataFrame({"A": [0, 0, 1, 1], "B": [1, 2, 3, 4]})
-        with pytest.raises(SpecificationError, match="Function names"):
-            df.groupby("A").agg(a=("A", "min"), b=("A", "min"))
+
+        grouped = df.groupby("A").agg(a=("B", "min"), b=("B", "min"))
+        expected = pd.DataFrame(
+            {"a": [1, 3], "b": [1, 3]}, index=pd.Index([0, 1], name="A")
+        )
+        tm.assert_frame_equal(grouped, expected)
 
```

Comment:
Can you also add a test using `functools.partial` as in the original issue?

Reply:
genius!! really have sharp eyes!! this will not raise an error, but will produce wrong result! It seems there is a bug in `_agg_1dim` in `base.py`, I will debug a bit and let you know in details. @TomAugspurger

Reply:
this might come down to having named aggregation in Series? i need time to look deeper in code base. Will update later.